simple/stats/reporter.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+from enum import auto
+from enum import Enum
+from functools import wraps
+import json
+import os
+import sys
+
+# For importing util
+_CODEDIR = os.path.dirname(os.path.realpath(__file__))
+sys.path.insert(1, os.path.join(_CODEDIR, "../"))
+
+from util.filehandler import FileHandler
+
+
+class Status(Enum):
+  NOT_STARTED = auto()
+  STARTED = auto()
+  SUCCESS = auto()
+  FAILURE = auto()
+
+
+class ImportReporter:
+  """Generates a report on every reported change.
+    
+    The report is written to report.json in the process directory.
+    """
+
+  def __init__(self, report_fh: FileHandler) -> None:
+    self.status = Status.NOT_STARTED
+    self.start_time = None
+    self.last_update = datetime.now()
+    self.report_fh = report_fh
+    self.data = {}
+    self.import_files: dict[str, FileImportReporter] = {}
+
+  def _report(func):
```

Comment:
Add a comment on the decorator function?

Reply:
Done.